tools/infer.py

```diff
@@ -82,22 +165,98 @@ def run(weights=osp.join(ROOT, 'yolov6s.pt'),
         hide_conf: Hide confidences
         half: Use FP16 half-precision inference, e.g. False
     """
+
+    # Initialize Weights & Biases
+
+    if not osp.isfile(weights) and not osp.isfile(weights + ".pt"):
+        try:
+            print("Downloading weights...")
+            weights = wget.download(
+                f"https://github.com/meituan/YOLOv6/releases/download/0.1.0/{weights}.pt"
+            )
+            print("\nDone.")
+        except urllib.error.HTTPError:
+            print("Unable to download model.")
+    elif osp.isfile(weights + ".pt"):
+        weights += ".pt"
+
+    if not osp.isfile(source) and not osp.isdir(source):
+        try:
+            print("Downloading image...")
+            source = wget.download(source)
+            print("\nDone.")
+        except urllib.error.HTTPError:
+            print("Unable to download image.")
+
+    if wandb_project is not None:
+        wandb.init(
```

Comment:
if this grows we'll have to figure out how to pass some more arguments to wandb.init